### PR TITLE
Cleanup vertical funnels breakdown

### LINKED ISSUE #6560: Showing/hiding other breakdown in funnels is buggy
## Bug description

When showing/hiding the "Other" property breakdown in funnels, all "Others" are selected which is confusing. Also, depending on ordering it may lead you to think that colors are wrong because you select a "Other" with a color but another color pops up.

image

Bonus, this general checkbox has a weird blue, should be the default primary color.

image

## Expected behavior

Only the line I show/hide should change.

## How to reproduce

See recording above.

## Environment

- [X] PostHog Cloud

### LINKED ISSUE #7275: Funnels - person modal can't be opened
## Bug description

On funnels (vertical), I can no longer click on the bar or label (e.g. “44 users”) to open the person modal. See user frustrated by this: [recording](https://app.posthog.com/recordings?sessionRecordingId=17d47134204ef1-04db7fe774d47f-3c700c58-1fa400-17d471342051070&filters=%7B%22actions%22%3A%5B%5D%2C%22events%22%3A%5B%7B%22id%22%3A%22%24pageview%22%2C%22type%22%3A%22events%22%2C%22order%22%3A0%2C%22name%22%3A%22%24pageview%22%2C%22properties%22%3A%5B%7B%22key%22%3A%22%24host%22%2C%22value%22%3A%5B%22posthog.com%22%5D%2C%22operator%22%3A%22is_not%22%2C%22type%22%3A%22event%22%7D%5D%7D%2C%7B%22id%22%3A%22%24rageclick%22%2C%22type%22%3A%22events%22%2C%22order%22%3A1%2C%22name%22%3A%22%24rageclick%22%7D%5D%2C%22properties%22%3A%5B%5D%2C%22date_from%22%3A%222021-10-23%22%2C%22date_to%22%3Anull%2C%22offset%22%3A50%2C%22session_recording_duration%22%3A%7B%22type%22%3A%22recording%22%2C%22key%22%3A%22duration%22%2C%22value%22%3A60%2C%22operator%22%3A%22gt%22%7D%7D&source=recordings_list); **01:09**.

## Expected behavior
Clicking on any of the red boxes above, should open the person modal.

## How to reproduce

1. Create a funnel.
2. Disable the vertical funnel FF (if enabled).
3. Try clicking within the redboxes shown above.

## Environment

- [x] PostHog Cloud
- [ ] self-hosted PostHog (ClickHouse-based), version/commit: _please provide_
- [ ] self-hosted PostHog (Postgres-based, legacy), version/commit: _please provide_

### LINKED ISSUE #7373: Inconsistent empty state in funnels
## Bug description
When I create a funnel that has zero persons, we show an empty funnel graph, instead of the default empty state for insights. Further, we show a drop in step 2 (visually), while both steps have zero persons.

## Expected behavior

Show the standardized empty error state.

## How to reproduce

1. Try [this funnel](https://app.posthog.com/insights/JpNtVpZb/edit?insight=FUNNELS&actions=%5B%5D&events=%5B%7B%22id%22%3A%22%24pageview%22%2C%22name%22%3A%22%24pageview%22%2C%22type%22%3A%22events%22%2C%22order%22%3A0%7D%2C%7B%22id%22%3A%22%24pageview%22%2C%22type%22%3A%22events%22%2C%22order%22%3A1%2C%22name%22%3A%22%24pageview%22%7D%5D&display=FunnelViz&interval=day&properties=%5B%7B%22key%22%3A%22action_name%22%2C%22value%22%3A%5B%22dont_exist%22%5D%2C%22operator%22%3A%22exact%22%2C%22type%22%3A%22event%22%7D%5D&funnel_to_step=1&funnel_viz_type=steps&exclusions=%5B%5D&funnel_from_step=0)

## Environment

- [ ] PostHog Cloud
- [ ] self-hosted PostHog (ClickHouse-based), version/commit: _please provide_
- [ ] self-hosted PostHog (Postgres-based, legacy), version/commit: _please provide_

### LINKED ISSUE #7409: Figure out funnel vertical breakdown
We have the `5733-funnel-vertical-breakdown` in an ambiguous state because it has some outstanding bugs to fix before we can ship it. We need to either fix those bugs and release it to everyone or scrap all these changes.

### PR DESCRIPTION
**Pull request by alexkim205**
## Changes

Fixes various vertical funnel breakdowns + other funnel-related bugs as part of Nail Insights QA sprint.

- [x] Fix how vertical breakdowns look on dashboards.
- [x] Fixes #7373
- [x] Fixes style bug where top to bottom layout padding was 0 when FF was on.
- [x] Fixes #7275
- [x] Fixes style bug where steps table (FF off) was rendered borderless.
- [x] Closes #7409
- [x] Fixes insight label weird '-' concatenation in funnel steps table.
- [x] Fixes #6560
	- [x] Multiple property breakdown inadvertently fixed this bug because we now return empty arrays for `breakdown_value` params. This is used to conditionally check if there is a valid breakdown type specified and create visibility keys for the legend. Although this fixes things, this broke typing in a lot of places, so this PR takes a first stab at leveling the playing field.
- [x] Fixes bug where person modal information was out of sync with UI. Added comment so we don't break it again.
- [x] Fixes small person modal styling bug with zeroed padding.
- [x] General refactor of breakdown components into separate file.

## State of FF

@paolodamico After this is merged, the `5733-funnel-vertical-breakdown` should be ready for rollout release. 

## How did you test this code?

Tested every combination of FF's. `5733-funnel-vertical-breakdown` x `5730-funnel-horizontal-ui` x `938-breakdown-by-multiple-properties` x `funnel-simple-mode`

No new logic was added other than a selector. Smoke tested bugs that were identified and changed.

## Vertical Breakdowns on Dashboards are fixed!

**Before**

Screen Shot 2021-11-30 at 3 59 31 PM

**After**

https://user-images.githubusercontent.com/13460330/144149811-aff44aec-ff2c-451f-9529-e6de1db4f33d.mov

## Tangentially related thing that was bugging me

It looks like with the new aggregation property in funnels we can show different units above the bar. There's a lot of repetition happening here. It also makes the empty space above each bar unclickable (to see dropoffs). Is there a better way to show the same information in a smarter way? @clarkus 

Screen Shot 2021-11-29 at 3 01 57 PM

Screen Shot 2021-11-29 at 3 03 39 PM

### CODE REVIEW

**Review by pauldambra — DISMISSED** [review]
The step index is out of sync between person modal and the UI. And has `- None` in the step name that doesn't match the main UI

out-of-syn-2

File: frontend/src/scenes/funnels/funnelUtils.ts [hunk 1]
```diff
@@ -255,6 +269,48 @@ export async function pollFunnel<T = FunnelStep[] | FunnelsTimeConversionBins>(
     return result
 }
 
+interface BreakdownStepValues {
+    rowKey: string
+    breakdown: (string | number)[]
+    breakdown_value: (string | number)[]
+}
+
+export const getBreakdownStepValues = (
```
[the remark this answers is not in the document]
Comment (alexkim205) [reply frontend/src/scenes/funnels/funnelUtils.ts:278]: That's a good idea, on it

**Comment by alexkim205** [comment]
> The step index is out of sync between person modal and the UI. And has `- None` in the step name that doesn't match the main UI

Definitely seems like another bug. I think this next revision should fix it.

File: frontend/src/scenes/funnels/funnelUtils.test.ts [hunk 2]
```diff
@@ -30,3 +30,72 @@ describe('getMeanAndStandardDeviation', () => {
...
+
+describe('getBreakdownStepValues()', () => {
+    it('is baseline breakdown', () => {
+        expect(getBreakdownStepValues({ breakdown: 'blah', breakdown_value: 'Blah' }, 21, true)).toStrictEqual({
+            rowKey: 'baseline_0',
+            breakdown: ['baseline'],
+            breakdown_value: ['Baseline'],
+        })
+    })
+    it('breakdowns are happy arrays', () => {
```
Comment (pauldambra) [inline frontend/src/scenes/funnels/funnelUtils.test.ts:42]: what's a "happy array"? I've not seen the term before...

File: frontend/src/scenes/funnels/funnelUtils.test.ts [hunk 3]
```diff
@@ -30,3 +30,72 @@ describe('getMeanAndStandardDeviation', () => {
         })
     })
 })
+
+describe('getBreakdownStepValues()', () => {
```
Comment (pauldambra) [inline frontend/src/scenes/funnels/funnelUtils.test.ts:34]: 👍

**Review by pauldambra — APPROVED** [review]
I wish these changes had been in before I worked on funnels :) 👍

File: frontend/src/scenes/funnels/funnelUtils.test.ts (hunk 2 above)
Comment (alexkim205) [reply frontend/src/scenes/funnels/funnelUtils.test.ts:42]: Haha I should probably call this "well formed"

Comment (pauldambra) [reply frontend/src/scenes/funnels/funnelUtils.test.ts]: Ah, I see. No, I love "happy" I'm going to start using it.

Knew someone who called braces, brackets, etc squares, smooths, and squiggles which has stuck with me as the easiest way to distinguish between them :)

**Pull request merged**